src/js/libcs/OpDrawing.js

```diff
@@ -107,18 +107,26 @@ evaluator.fillcircle$2 = function(args, modifs) {
 };
 
 evaluator.drawarc$3 = function(args, modifs) {
-    return eval_helper.drawarc(args, modifs, "D");
+    var newargs = [csgeo.csnames[args[0].name].homog,
+        csgeo.csnames[args[1].name].homog,
+        csgeo.csnames[args[2].name].homog
+    ];
+    return eval_helper.drawarc(newargs, modifs, "D");
 };
 
 evaluator.fillarc$3 = function(args, modifs) {
-    return eval_helper.drawarc(args, modifs, "F");
+    var newargs = [csgeo.csnames[args[0].name].homog,
+        csgeo.csnames[args[1].name].homog,
+        csgeo.csnames[args[2].name].homog
+    ];
+    return eval_helper.drawarc(newargs, modifs, "F");
 };
 
```

Comment:
No good! Here you're assuming that the arguments will be geometric points. Just like it was before 3c550cd2b25f78d28714a5de9b46c152a4932ee4. Here you are at the CindyJS level, and `evaluateAndHomog` is all you need.